crates/btcio/src/reader/query.rs

```diff
@@ -299,28 +304,129 @@ pub async fn get_verification_state(
     let vh = height - 1; // verified_height
     let vb = client.get_block_at(vh).await?; // verified_block
 
-    // Fetch the previous timestamps of block from `vh`
-    // This fetches timestamps of `vh`, `vh-1`, `vh-2`, ...
     const N: usize = 11;
-    let mut timestamps: [u32; 11] = [0u32; 11];
-    for i in (0..N).rev() {
-        if vh > i as u64 {
-            let h = client.get_block_at(vh - i as u64).await?;
-            timestamps[i] = h.header.time;
+    let mut timestamps: [u32; N] = [0u32; N];
+
+    // Fetch the previous timestamps of block from `vh`
+    // This fetches timestamps of `vh-10`,`vh-9`, ... `vh-1`, `vh`
+    for i in 0..N {
+        if vh >= i as u64 {
+            let height_to_fetch = vh - i as u64;
+            let h = client.get_block_at(height_to_fetch).await?;
+            timestamps[N - 1 - i] = h.header.time;
         } else {
-            timestamps[i] = 0;
+            // No more blocks to fetch; the rest remain zero
+            timestamps[N - 1 - i] = 0;
```

Comment:
Curious about what's changing here.  And couldn't we load this out of the db instead of fetching it again?

Reply:
It turns out there was error on how the timestamps were ordered previously. This fixes that. Now, timestamps[0] will be the timestamp of the oldest block. 
We can load this from the db. Right now this is called only once by the sequencer for the genesis block. The HeaderVerificationState is then in the client state and is updated there. This is also called from the prover-client independently right now. We want to fetch the HeaderVerificationState directly from the sequencer later on.